protocol/communities/permission_checker.go

```diff
@@ -33,6 +33,19 @@ type DefaultPermissionChecker struct {
 	logger *zap.Logger
 }
 
+type PreParsedPermissionsData struct {
+	Erc20TokenRequirements  map[uint64]map[string]*protobuf.TokenCriteria
+	Erc721TokenRequirements map[uint64]map[string]*protobuf.TokenCriteria
+	Erc20TokenAddresses     []gethcommon.Address
+	Erc20ChainIDsMap        map[uint64]bool
```

Comment:
```suggestion
	Erc20ChainIDsMap        map[uint64]struct{}
```

It is more idiomatic in go to use `map[uint64]struct{}` for sets because it clearly indicates that only the presence of keys matters.

Reply:
I think it was done to have a shorter code for the case:
```
func calculateChainIDsSet(accountsAndChainIDs []*AccountChainIDsCombination, requirementsChainIDs map[uint64]bool) []uint64 {

	revealedAccountsChainIDs := make([]uint64, 0)
	revealedAccountsChainIDsMap := make(map[uint64]bool)

	// we want all chainIDs provided by revealed addresses that also exist
	// in the token requirements
	for _, accountAndChainIDs := range accountsAndChainIDs {
		for _, chainID := range accountAndChainIDs.ChainIDs {
                         // Here you don't need to do _, exists := requirementsChainIDs[chainID]
			if requirementsChainIDs[chainID] && !revealedAccountsChainIDsMap[chainID] {
				revealedAccountsChainIDsMap[chainID] = true
				revealedAccountsChainIDs = append(revealedAccountsChainIDs, chainID)
			}
		}
	}
	return revealedAccountsChainIDs
}
```

So I assume it is ok to have map[uint64]bool